src/adapters/postgres/cohort-adapter.ts

```diff
@@ -135,12 +138,44 @@ export class PostgresCohortService {
         options: data?.fieldParams?.['options'] || {},
         type: data.type || ''
       };
+      if(data.source_details){
+        // We need to add teh dependence Condition here.
+        if(data?.source_details?.source === 'table'){ 
+          let dynamicOptions = await this.findDynamicOptions(data?.source_details?.table);
+          customField.options = dynamicOptions
+        }else if(data.source_details.source === 'jsonFile'){
+          // let findDataFromJson = 
+        }else{
+          customField.options = data.fieldParams;
+        }
+      }
       customFieldsArray.push(customField);
     }
     result.cohortData['customFields'] = customFieldsArray;
     return result
   }
 
+  async findDynamicOptions(tableName,whereClause?:{}){
```

Comment:
@Shubham4026 @souravbhowmik1999 This is field specific code, should be in field adapter